### PR TITLE
feat(sdk-core): ecdsa signing with static trusted challenges

### PR DESCRIPTION
**Pull request by zahin-mohammad**
## Description

This PR introduces support for ECDSA signing using static enterprise challenges if the enterprise has the specified feature flag. Currently both the existing flow and the new signing flow will continue to work until the feature flag is deprecated. 

## Issue Number

TICKET: BG-75009

## Type of change

Please delete options that are not relevant.

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update

# How Has This Been Tested?

- Existing unit tests. 
- Manual testing in testnet-02 (regression testing and the new flow)

# Checklist:

- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [x] My code compiles correctly for both Node and Browser environments
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] My commits follow [Conventional Commits](https://www.conventionalcommits.org/en/v1.0.0/) and I have properly described any BREAKING CHANGES
- [x] The ticket or github issue was included in the commit message as a reference
- [x] I have made corresponding changes to the documentation and on any new/updated functions and/or methods - [jsdoc](https://jsdoc.app/)
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] New and existing unit tests pass locally with my changes

### CODE REVIEW

File: modules/sdk-core/src/account-lib/mpc/tss/ecdsa/ecdsa.ts [hunk 1]
```diff
@@ -36,10 +35,61 @@ import {
...
+  SerializedNtilde,
+  DeserializedNtilde,
 } from './types';
 
 const _5n = BigInt(5);
 
+/**
+ * Deserializes a challenge and it's proofs from hex strings to bigint
+ */
+export function deserializeNtilde(challenge: SerializedNtilde): DeserializedNtilde {
```
Comment (zahin-mohammad) [inline modules/sdk-core/src/account-lib/mpc/tss/ecdsa/ecdsa.ts:47]: Moved due to circular dependencies. Also makes sense to keep this closer to the types file that defines these two types.

File: modules/sdk-core/src/account-lib/mpc/tss/ecdsa/ecdsa.ts [hunk 2]
```diff
@@ -291,21 +341,15 @@ export default class Ecdsa {
...
    * this signing operation
-   * @param challenge
+   * @param {SerializedNtilde} challenge
    * @returns {KeyCombined} The new XShare and YShares with the amended
    * challenge values
    */
-  async appendChallenge(
-    xShare: XShare,
-    yShare: YShare,
-    challenge?: DeserializedNtilde
```
Comment (zahin-mohammad) [inline modules/sdk-core/src/account-lib/mpc/tss/ecdsa/ecdsa.ts:301]: It was odd to accept the `DeserializedNtilde` as input instead of a `SerializedNtilde`. This was likely due to the circular dependency pointed out above.

File: modules/sdk-core/src/bitgo/bitgoBase.ts [hunk 3]
```diff
@@ -23,5 +23,4 @@ export interface BitGoBase {
   setRequestTracer(reqTracer: IRequestTracer): void;
   url(path: string, version?: number): string;
   register(name: string, coin: CoinConstructor): void;
-  getChallengesForEcdsaSigning(walletId: string): Promise<any>;
```
Comment (zahin-mohammad) [inline modules/sdk-core/src/bitgo/bitgoBase.ts:26]: Moved to the wallet class.

File: modules/sdk-core/src/bitgo/enterprise/enterprise.ts [hunk 4]
```diff
@@ -29,6 +31,7 @@ export class Enterprise implements IEnterprise {
     if (!_.isString(enterpriseData.name)) {
       throw new Error('enterprise name has to be a string');
     }
+    this._enterprise = enterpriseData;
```
Comment (zahin-mohammad) [inline modules/sdk-core/src/bitgo/enterprise/enterprise.ts:34]: This is the same pattern used in the wallet class. It was odd to see this only half completed for this class.

File: modules/sdk-core/src/bitgo/enterprise/iEnterprise.ts [hunk 5]
```diff
@@ -4,6 +4,12 @@ import { Buffer } from 'buffer';
 import { BitGoProofSignatures } from '../utils/tss/ecdsa';
 import { DeserializedNtilde } from '../../account-lib/mpc/tss/ecdsa/types';
 
+export interface EnterpriseData {
+  id: string;
+  name: string;
+  featureFlags?: string[];
+}
```
Comment (zahin-mohammad) [inline modules/sdk-core/src/bitgo/enterprise/iEnterprise.ts:11]: Fully typing this model is out of scope in this PR.

File: modules/sdk-core/src/bitgo/utils/tss/ecdsa/ecdsa.ts [hunk 6]
```diff
@@ -647,7 +650,7 @@ export class EcdsaUtils extends baseTSSUtils<KeyShare> {
           ? txRequestResolved.transactions![0].unsignedTx
           : txRequestResolved.unsignedTxs[0];
       signablePayload = Buffer.from(unsignedTx.signableHex, 'hex');
-      derivationPath = txRequestResolved.transactions![0].unsignedTx.derivationPath;
+      derivationPath = unsignedTx.derivationPath;
```
Comment (zahin-mohammad) [inline modules/sdk-core/src/bitgo/utils/tss/ecdsa/ecdsa.ts:653]: This would have caused an error if txRequest lite was used!

File: modules/sdk-core/src/bitgo/enterprise/iEnterprise.ts [hunk 7]
```diff
@@ -23,4 +29,5 @@ export interface IEnterprise {
     challenge?: DeserializedNtilde
   ): Promise<void>;
   getExistingTssEcdsaChallenge(): Promise<DeserializedNtilde>;
+  hasFeatureFlags(flags: string[]): boolean;
```
Comment (zahin-mohammad) [inline modules/sdk-core/src/bitgo/enterprise/iEnterprise.ts:32]: TODO: type the feature flag enum

File: modules/sdk-core/src/bitgo/utils/tss/ecdsa/ecdsa.ts [hunk 8]
```diff
@@ -988,55 +1020,6 @@ export class EcdsaUtils extends baseTSSUtils<KeyShare> {
     return await bitgo.get(bitgo.url('/tss/ecdsa/challenges', 2)).send().result();
   }
 
-  /**
-   * Deserializes a challenge and it's proofs from hex strings to bigint
-   */
-  static deserializeNtilde(challenge: SerializedNtilde): DeserializedNtilde {
```
Comment (zahin-mohammad) [inline modules/sdk-core/src/bitgo/utils/tss/ecdsa/ecdsa.ts:994]: Moved (see comments above)

File: modules/express/encryptedPrivKeys.json [hunk 9]
```diff
@@ -1,3 +1,3 @@
 {
-	"61f039aad587c2000745c687373e0fa9": "{\"iv\":\"+4Rbtj5f1XCG/aCGEvT36Q==\",\"v\":1,\"iter\":10000,\"ks\":256,\"ts\":64,\"mode\":\"ccm\",\"adata\":\"\",\"cipher\":\"aes\",\"salt\":\"Xa8h0OwvxrM=\",\"ct\":\"R2UBZeyz5jvP937w17/xAOX4Bw1/PQ2ttQD53Pa2kr1lsaL/zcXv/av5HOPKastjRhyOeT7kmG2NpyKr0q7jxByuQSLRYEgLuE5quDZUrweQxQF2GYnOVN+569s7kpkLKss888y4yudPn68p3T+/L+KGoO0r8GQ=\"}"
```
Comment (zahin-mohammad) [inline modules/express/encryptedPrivKeys.json:2]: This was updated while running all the unit tests locally.

File: modules/sdk-core/src/bitgo/enterprise/iEnterprise.ts [hunk 10]
```diff
@@ -4,6 +4,14 @@ import { Buffer } from 'buffer';
 import { BitGoProofSignatures } from '../utils/tss/ecdsa';
 import { DeserializedNtilde } from '../../account-lib/mpc/tss/ecdsa/types';
 
+export type EnterpriseFeatureFlag = 'useEnterpriseEcdsaTssChallenge';
```
Comment (zahin-mohammad) [inline modules/sdk-core/src/bitgo/enterprise/iEnterprise.ts:7]: Fully typing all options is out of scope for this PR

**Review by pranavjain97 — APPROVED** [review]
